### PR TITLE
fix(source:push): support pushing large documents 

### PR DESCRIPTION
**Pull request by y-lakhdar**
https://coveord.atlassian.net/browse/CDX-710

## Proposed changes

Small change, big impact!

If a document size was greater than 5MB (max content length), an empty batch was uploaded to the API.
This was creating a significant amount of "empty batch" requests (see image).

### Example
Here, I am pushing 2 documents of 9 MB each.

#### Before change
Since both documents exceed the maxContentLength limit, the `accumulator` is always empty. Therefore, 2 initial (empty) batches are sent by the `send` method. 
Finally, the `close` method upload another batch with only one document
image

#### After change
We do not want to upload a batch if it is empty
image

## Testing

- [x] Unit Tests:

- [ ] Functionnal Tests:

- [x] Manual Tests:

### CODE REVIEW

**Review by olamothe — APPROVED** [review]

**Review by louis-bompart — CHANGES_REQUESTED** [review]
Can we add tests to cover this use case?

File: packages/cli/src/commands/source/push/add.spec.ts [hunk 1]
```diff
@@ -170,6 +176,26 @@ describe('source:push:add', () => {
       expect(ctx.stdout).toContain('Status code: 202 👌');
     });
 
+  test
+    .do(() => {
+      const largeFileSize = 5 * 1024 * 1024;
```
Comment (louis-bompart) [inline packages/cli/src/commands/source/push/add.spec.ts:181]: So 5 mebibytes ?
If you mean 5 megabytes, it's 5e6

**Review by louis-bompart — APPROVED** [review]
LGTM, double-check if the size limits are in metric megabytes or in legacy megabytes aka mebibytes and adapt the tests accordingly

File: packages/cli/src/commands/source/push/add.spec.ts (hunk 1 above)
Comment (y-lakhdar) [reply packages/cli/src/commands/source/push/add.spec.ts:181]: nice

**Pull request merged**